pkg/webhook/validation.go

```diff
@@ -164,24 +164,6 @@ func (v *validator) validate(obj runtime.Object) (admission.Warnings, error) {
 		))
 	}
 
-	if target := bundle.Spec.Target.ConfigMap; target != nil {
-		path := path.Child("sources")
-		for i, source := range bundle.Spec.Sources {
-			if source.ConfigMap != nil && source.ConfigMap.Name == bundle.Name && source.ConfigMap.Key == target.Key {
```

Comment:
We don't just check that the keys match, we also check that the bundle name matches the source config map name.

Reply:
Thanks for pointing this out, @inteon. I will make an effort to fix the currently broken validation smarter.